Library/test-helpers/lib.sh

```diff
@@ -1208,19 +1232,27 @@ Returns 0 when the start was successful, 1 otherwise.
 
 limeWaitForAgentRegistration() {
     local TIMEOUT=${limeTIMEOUT}
+    if [ "$1" == "--container" ]; then
+        local NAME="$2"
+        shift 2
+    fi
     local UUID="$1"
     local OUTPUT=`mktemp`
     [ -z "$1" ] && return 3
     [ -n "$2" ] && TIMEOUT=$2
 
     for I in `seq $TIMEOUT`; do
-        keylime_tenant -c regstatus -u $UUID &> $OUTPUT
-	if grep -q "Agent $UUID exists on registrar" $OUTPUT; then
+        if [ -n "$NAME" ]; then
+            podman exec -i $NAME keylime_tenant -c regstatus -u $UUID &> $OUTPUT
```

Comment:
I guess there is one thing we understand differently. In your case tenant container is a "service", a long-running process that is being used to query verifier/registrar eventually. 
In my understanding this is a one-shot container, that is started whenever we need to call keylime_tenant and ends rigth after it. IMHO there is no need to keep it running for a long time. 
This misunderstanding projects into the whole code. Probably we should discuss it in person.

Reply:
DONE.